From 19d74f6a88adb3cd1294e55000850eef80d743a3 Mon Sep 17 00:00:00 2001
From: Sunil Kumar Kori <skori@marvell.com>
Date: Tue, 3 Jan 2023 15:00:20 +0530
Subject: [PATCH 121/955] common/cnxk: reduce channel count per LMAC

Due to limitation, imposed by latest kernel, maximum number of
channel supported per LMAC is reduced to 8. Because of this change
application gets failed to initialize for more channels.

Also this limitation impacts PFC functional behaviour.

So patch just aligns the supported number of channel with kernel.

Signed-off-by: Sunil Kumar Kori <skori@marvell.com>
Change-Id: I3f3fac0bd9dac99a91c17cad5cb25ee375f82c4d
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/93718
Reviewed-by: Nithin Kumar Dabilpuram <ndabilpuram@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 drivers/common/cnxk/roc_mbox.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/common/cnxk/roc_mbox.h b/drivers/common/cnxk/roc_mbox.h
index 92a29f4446134..ca2f845ec6f7e 100644
--- a/drivers/common/cnxk/roc_mbox.h
+++ b/drivers/common/cnxk/roc_mbox.h
@@ -1174,7 +1174,7 @@ struct nix_bp_cfg_req {
  * so maximum 256 channels are possible.
  */
 #define NIX_MAX_CHAN	 256
-#define NIX_CGX_MAX_CHAN 16
+#define NIX_CGX_MAX_CHAN 8
 #define NIX_LBK_MAX_CHAN 1
 struct nix_bp_cfg_rsp {
 	struct mbox_msghdr hdr;
-- 
2.25.1

